### PR TITLE
vector: include the "unix" cargo feature if targetPlatform isUnix

### PR DESCRIPTION
**Pull request by johanot**
###### Motivation for this change

Fixes the issue that @happysalada is facing here: https://github.com/NixOS/nixpkgs/pull/103393#issuecomment-726434830

The journald source is apparently guarded by the "unix" cargo feature flag:
https://github.com/timberio/vector/blob/v0.10.0/src/sources/mod.rs#L14-L15

Don't know why the Vector folks made the choice that unix is a "feature". There might be a good reason why it's not just: `#[cfg(unix)]`, but never mind. Including "unix" in the default features when targetPlatform is unix should be sufficient for nixpkgs.

cc @thoughtpolice 

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Hoverbear** [comment]
Ah-ha! Of course. This is 0.10. We redid our release process for 0.11. https://github.com/timberio/vector/blob/a46a2e488d353eff2b476fd9984304cbe2f67424/src/sources/mod.rs#L14-L15

So yes, for 0.10 you'll need unix here to get the `journald` source. For 0.11 you can remove this feature.

**Comment by johanot** [comment]
> For 0.11 you can remove this feature.

@Hoverbear Right. :) I've made an alternative suggestion for a tempfix here: d1f1a88c469dec7e89d7e92ae6c9db52a2a77e41. I'll let the reviewer decide which is best. With fetchpatch, we won't forget to cleanup, since the patch will fail when already applied.

**Comment by happysalada** [comment]
@johanot @Hoverbear Amazing!
If you are not too busy with life, consider adding yourself as a maintainer of this package :-).

**Review by thoughtpolice — APPROVED** [review]

**Comment by thoughtpolice** [comment]
Thanks everyone!

**Pull request merged**
